simple_parsing/wrappers/field_wrapper.py

```diff
@@ -1040,3 +1028,87 @@ def only_keep_action_args(options: Dict[str, Any], action: Union[str, Any]) -> D
         logger.debug(f"Kept options: \t{kept_options.keys()}")
         logger.debug(f"Removed options: \t{deleted_options.keys()}")
     return kept_options
+
+
+_BOOL_NO_PREFIX = "--no"
+
+
+class _BooleanOptionalAction(argparse.Action):
+    """Similar to `argparse.BooleanOptionalAction`.
+
+    * Support custom prefix (for `absl.flags` compatibility)
+    * Accept `--flag=true` value
+    * Support `nargs`
+    * Support Python 3.8
+
+    """
+
+    def __init__(
+        self,
+        option_strings,
+        dest,
+        default=None,
+        type=None,
+        choices=None,
+        required=False,
+        help=None,
+        metavar=None,
+        nargs=None,
+    ):
+        if nargs is None:
+            nargs = '?'
+        if not (isinstance(nargs, int) or nargs in {'?', '*', '+'}):
+            raise ValueError(f'nargs={nargs!r} not supported for bools')
+
+        # Do not add `--no` when `nargs >= 1`
+        if nargs == '+' or (isinstance(nargs, int) and nargs >= 1):
+            new_option_strings = option_strings
+        else:
+            new_option_strings = []
+            for option_string in option_strings:
+                new_option_strings.append(option_string)
+
+                if option_string.startswith("--"):
+                    option_string = _BOOL_NO_PREFIX + option_string[2:]
+                    new_option_strings.append(option_string)
+
+        super().__init__(
+            option_strings=new_option_strings,
+            dest=dest,
+            nargs=nargs,
+            default=default,
+            type=type,
+            choices=choices,
+            required=required,
+            help=help,
+            metavar=metavar,
+        )
+
+    def __call__(
+        self,
+        parser: argparse.ArgumentParser,
+        namespace: argparse.Namespace,
+        values: Any,
+        option_string=None,
+    ):
+        if option_string in self.option_strings:
+            is_neg = option_string.startswith(_BOOL_NO_PREFIX)
+            if values is None:  # --my_flag / --nomy_flag
+                bool_value = not is_neg
+            elif values == []:  # --my_flag / --nomy_flag (with nargs='*')
+                bool_value = [not is_neg]
+            elif is_neg:
+                parser.exit(
+                    message=f"{_BOOL_NO_PREFIX} cannot be used with value (Got: {option_string}={values})"
+                )
+            elif isinstance(values, str):  # --my_flag true
+                bool_value = utils.str2bool(values)
+            elif isinstance(values, list):  # --my_flag true true false
+                bool_value = [utils.str2bool(v) for v in values]
+            else:
+                raise ValueError(f"Unsuported value for {option_string}: {values!r}")
+
+            setattr(namespace, self.dest, bool_value)
+
+    def format_usage(self):
```

Comment:
Would this usage str also be correct when using `nargs`?

Reply:
`--noflag` is only added when `nargs == '*' '?'` or `nargs >= 1`, so it won't appear in the help otherwise.

Reply:
I see. Makes sense.